account.rs/zk-rs/src/poseidon.rs

```diff
@@ -0,0 +1,73 @@
+use ark_bn254::Fr;
+use js_sys::{Array, Uint8Array};
+use light_poseidon::{Poseidon, PoseidonBytesHasher, PoseidonError};
+use num_bigint::BigUint;
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
+
+use crate::utils::set_panic_hook;
+
+#[wasm_bindgen]
+pub fn poseidon(inputs: &Array) -> Result<Uint8Array, JsValue> {
+    set_panic_hook();
+    let len = inputs.length();
+    let mut inputs_vec: Vec<Vec<u8>> = Vec::new();
+    for index in 0..len {
```

Comment:
Is this the only way to iterate over `Array`? It seems like it has an `iter()` method:

https://rustwasm.github.io/wasm-bindgen/api/js_sys/struct.Array.html#method.iter

If that works, we could even try to use `iter().map` and `collect` (it's faster than manual loops).